cloudconfig/containerinit/container_userdata_test.go

```diff
@@ -187,6 +187,31 @@ end script
 	testing.CheckWriteFileCommand(c, cmds[0], filename, script, nil)
 }
 
+func (s *UserDataSuite) TestShutdownInitCommandsUpstartDHCP(c *gc.C) {
+	cmds, err := containerinit.ShutdownInitCommands(service.InitSystemUpstart, "trusty")
+	c.Assert(err, jc.ErrorIsNil)
+
+	filename := "/etc/init/juju-template-restart.conf"
+	script := `
+description "juju shutdown job"
+author "Juju Team <juju@lists.ubuntu.com>"
+start on stopped cloud-final
+
+script
+  /sbin/dhclient -r $(/usr/bin/awk '/^iface.*dhcp$/ {print $2}' /etc/network/interfaces /etc/network/interfaces.d/*) &&
+  /sbin/ifdown $(/usr/bin/awk '/^iface.*dhcp$/ {print $2}' /etc/network/interfaces /etc/network/interfaces.d/*)
+  /bin/rm -fr /var/lib/dhcp/dhclient* /var/log/cloud-init*.log
```

Comment:
I don't see these logs as relevant to the cloned container, and see them as more likely confusing than useful. I'm not sure why they have been treated differently to the address-allocation case.

In both cases, they could be moved rather than deleted, but I don't think they should be left in place.